tests/lib/ifaces/linux_bridge_iface_test.py

```diff
@@ -0,0 +1,420 @@
+from ..testlib.bridgelib import SLAVE2_PORT_CONFIG
+from ..testlib.bridgelib import SLAVE2_VLAN_CONFIG_TRUNK
+from ..testlib.bridgelib import TRUNK_TAGS_ID_RANGES
+from ..testlib.bridgelib import gen_bridge_iface_info
+from ..testlib.bridgelib import gen_bridge_iface_info_with_vlan_filter
+from ..testlib.ifacelib import gen_foo_iface_info
+
+Port = LB.Port
+Vlan = LB.Port.Vlan
+
+
+class TestLinuxBridgeIface:
+    def test_linux_bridge_sort_slaves(self):
+        iface_info1 = gen_bridge_iface_info()
+        iface_info2 = gen_bridge_iface_info()
+        iface_info2[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE].reverse()
+
+        assert (
+            LinuxBridgeIface(iface_info1).to_dict()
+            == LinuxBridgeIface(iface_info2).to_dict()
+        )
+
+    def test_is_master(self):
+        iface = LinuxBridgeIface(gen_bridge_iface_info())
+
+        assert iface.is_master
+
+    def test_is_virtual(self):
+        iface = LinuxBridgeIface(gen_bridge_iface_info())
+
+        assert iface.is_virtual
+
+    def test_get_slaves(self):
+        iface = LinuxBridgeIface(gen_bridge_iface_info())
+
+        assert iface.slaves == TEST_SLAVE_NAMES
+
+    def test_desire_port_name_full_merge_from_current(self):
+        cur_iface_info = gen_bridge_iface_info()
+        des_iface_info = gen_bridge_iface_info()
+        des_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            {Port.NAME: SLAVE2_IFACE_NAME},
+            {Port.NAME: SLAVE1_IFACE_NAME},
+        ]
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == gen_bridge_iface_info()
+
+    def test_desire_port_name_partial_merge_from_current(self):
+        cur_iface_info = gen_bridge_iface_info()
+        des_iface_info = gen_bridge_iface_info()
+        expected_iface_info = gen_bridge_iface_info()
+        expected_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            deepcopy(SLAVE2_PORT_CONFIG)
+        ]
+        des_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            {Port.NAME: SLAVE2_IFACE_NAME},
+        ]
+
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == expected_iface_info
+
+    def test_merged_with_desire_remove_all_ports(self):
+        cur_iface_info = gen_bridge_iface_info()
+        des_iface_info = gen_bridge_iface_info()
+        expected_iface_info = gen_bridge_iface_info()
+        expected_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = []
+        des_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = []
+
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == expected_iface_info
+
+    def test_merge_with_desire_port_not_defined(self):
```

Comment:
typo on the test name; `test_merge_with_desireD_port_not_defined`. I'd also use plural `ports`, since that is more representative of the port subtree cardinality.

Reply:
Changed to `test_merge_with_desired_ports_not_defined`.